### PR TITLE
Check duplicate `depends_on` of `Plugin`

### PR DESCRIPTION
**Pull request by yuema137**
### Explanation
Basically what happened was:
- In `plugin.py`, the chunks are fetched for each `self.depends_on` and stored in a dictionary: https://github.com/AxFoundation/strax/blob/2218e78722cb64cdedb17e255752b7857c5f229b/strax/plugins/plugin.py#L458-L493
- Therefore, when there are duplicate items in `self.depends_on` , the later one overwrites the previous one
- The later one cannot get anything, because the fetch is done in this way: https://github.com/AxFoundation/strax/blob/2218e78722cb64cdedb17e255752b7857c5f229b/strax/plugins/plugin.py#L467-L468 As the data is already read and split by the previous fetch, the later one can only get an empty result
- So finally both ones are empty

### Solution
The fix is simple, which is to remove the duplicate items in `self.depends_on` of the `Plugin` class.

### CODE REVIEW

File: strax/plugins/plugin.py [hunk 1]
```diff
@@ -110,6 +110,8 @@ def __init__(self):
             raise ValueError(f"depends_on not provided for {self.__class__.__name__}")
 
         self.depends_on = strax.to_str_tuple(self.depends_on)
+        # Remove duplicates
+        self.depends_on = tuple(dict.fromkeys(self.depends_on))
```
Comment (dachengx) [inline strax/plugins/plugin.py:114]: Better use `tuple(set(self.depends_on))`.

**Review by dachengx — CHANGES_REQUESTED** [review]

File: strax/plugins/plugin.py (hunk 1 above)
Comment (yuema137) [reply strax/plugins/plugin.py]: I believe it's the same functionally, but yes, let's make it easier.

**Comment by yuema137** [comment]
@dachengx In order to pass the pre-commit check I had to format it with black & flake. I think it happens now and then when we make PR for the files that were not checked previously. Should we do a through cleaning for strax?

File: strax/plugins/plugin.py [hunk 2]
```diff
@@ -607,7 +620,8 @@ def do_compute(self, chunk_i=None, **kwargs):
                 # This block will be deleted </end>
             elif len(set(tranges.values())) != 1:
                 message = (
-                    f"{self.__class__.__name__} got inconsistent time ranges of inputs: {tranges}"
+                    f"{self.__class__.__name__} "
```
Comment (dachengx) [inline strax/plugins/plugin.py:623]: Why this change is needed?

File: strax/plugins/plugin.py [hunk 3]
```diff
@@ -459,14 +461,25 @@ class IterDone(Exception):
                     # Fetch other inputs (when needed)
                     for d in self.depends_on:
                         if d != pacemaker:
+                            print(
```
Comment (dachengx) [inline strax/plugins/plugin.py:464]: Please remove these prompt.

File: strax/plugins/plugin.py [hunk 4]
```diff
@@ -110,6 +110,8 @@ def __init__(self):
             raise ValueError(f"depends_on not provided for {self.__class__.__name__}")
 
         self.depends_on = strax.to_str_tuple(self.depends_on)
+        # Remove duplicates
+        self.depends_on = tuple(self.depends_on)
```
Comment (dachengx) [inline strax/plugins/plugin.py:114]: `self.depends_on = tuple(set(self.depends_on))`

**Review by dachengx — CHANGES_REQUESTED** [review]

File: strax/plugins/plugin.py (hunk 4 above)
Comment (dachengx) [reply strax/plugins/plugin.py:114]: But will this really kill the issue? https://github.com/AxFoundation/strax/issues/846

File: strax/plugins/plugin.py (hunk 2 above)
Comment (yuema137) [reply strax/plugins/plugin.py:623]: the line is to long and black complains

**Review by dachengx — APPROVED** [review]
